### PR TITLE
run val only if val dataloader exists

### PR DESCRIPTION
**Pull request by gshennvm**
# What does this PR do ?

aligner models have no val dataloader, so we need not to reconfigure val micro batches

### CODE REVIEW

**Comment by gshennvm** [comment]
jenkins

File: nemo/collections/nlp/models/language_modeling/megatron_base_model.py [hunk 1]
```diff
@@ -333,8 +333,8 @@ def _reconfigure_val_batches(self):
             self.trainer.limit_val_batches *= get_num_microbatches()
         else:
             assert isinstance(self.trainer.limit_val_batches, float)
-            # Don't reconfigure if limit_val_batches is 0.0
-            if self.trainer.limit_val_batches == 0.0:
+            # Don't reconfigure if limit_val_batches is 0.0 or if there's no val dataloader
+            if self.trainer.limit_val_batches == 0.0 or self._validation_dl is None:
                 return
```
[the remark this answers is not in the document]
Comment (gshennvm) [reply nemo/collections/nlp/models/language_modeling/megatron_base_model.py:338]: i added it on this line just to save space. the logic here is orthogonal to limit val batches.

aligner doesn't use the standard PTL loop, so we don't have the limit val batches issue you have to fix

**Comment by athitten** [comment]
jenkins

**Review by athitten — APPROVED** [review]
LGTM, thanks!

**Comment by athitten** [comment]
jenkins

**Pull request merged**